text/0000-std-aware-cargo.md

```diff
@@ -0,0 +1,289 @@
+* Only compile time features considered `stable` may be used outside of nightly. Initially the list of `stable` features would be empty, and stabilizing these features would require a PR/RFC to `libstd`.
+
+### User Interaction
+
+The building of `std` would respect the current build profile, including
+
+The syntax for these features would look as follows:
+
+```toml
+[dependencies.std]
+default-features = false
+features = [
+    "force_alloc_system",
+]
+```
+
+It is not necessary to explicitly mention the dependency of `std`, unless changes to features are necessary.
+
+### Technical Implications
+
+None beyond the technical implications listed for `core`.
+
+## 4 - Allow the user to provide their own custom source versions of `core` and `std`
+
+### Use Case
+
+This will allow users of a nightly compiler to provide a custom version of `core` and `std`, without requiring the recompilation of the compiler itself.
+
+### Caveats
+
+As stability guarantees cannot be made around modified versions of `core` or `std`, a nightly compiler must always be used.
+
+### User Interaction
+
+For this interaction, the existing `patch` syntax of Cargo.toml will be used. For example:
+
+```toml
+[patch.sysroot]
+core = { path = 'my/local/core' }
+std  = { git = 'https://github.com/example/std' }
+```
+
+### Technical Implications
+
+The `patch.sysroot` term will be introduced for patch when referring to components such as `std` and `core`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This RFC introduces new concepts to the use of Rust and Cargo, and could be confusing for current users of Rust who have not had to consider changes to `core` or `std` previously. However, in the normal case, most users are unlikely to need these settings, while they allow users that DO need to make changes to control important steps of the build process.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+> Why is this design the best in the space of possible designs?
+
+This approach borrows from existing behaviors used by Cargo to allow configuration of `core` and `std`, as if they were a regular crate dependency.
+
+This approach also offers an approach that can be developed and applied incrementally, allowing for time to find coner cases not considered by this RFC
+
+> What other designs have been considered and what is the rationale for not choosing them?
+
+To the author of this RFC's knowledge, there are no other open designs, other than the use tools that wrap Cargo entirely, such as [xargo].
+
+[xargo]: https://github.com/japaric/xargo
+
+> What is the impact of not doing this?
+
+By not doing this, Rust will continue to be difficult to use for users and platforms "on the edge", such as new platform developers or embedded and WASM users.
+
+# Prior art
+[prior-art]: #prior-art
+
+* https://github.com/rust-lang/rfcs/pull/1133
+* https://github.com/japaric/xargo
```

Comment:
You may want to elaborate on these links and give a summary :)

Reply:
https://github.com/rust-lang/cargo/pull/2768 you might also want to link and also https://github.com/rust-lang/cargo/issues/5002 and https://github.com/rust-lang/cargo/issues/5003